ListUtil.xs

```diff
@@ -1217,12 +1221,33 @@ CODE:
 #endif
             }
 
-            if(!SvOK(arg) || SvUOK(arg))
+            if(!SvOK(arg) || SvUOK(arg)) {
                 sv_setpvf(keysv, "%" UVuf, SvUV(arg));
-            else if(SvIOK(arg))
+            }
+            else if(SvIOK(arg)) {
                 sv_setpvf(keysv, "%" IVdf, SvIV(arg));
-            else
-                sv_setpvf(keysv, "%.15" NVgf, SvNV(arg));
+            }
+            else {
+                nv_arg = SvNV(arg);
+                /* use 0 for both 0 and -0.0 */
+                if(nv_arg == 0) {
+                    sv_setpvs(keysv, "0");
+                }
+                /* for NaN, use the platform's normal stringification */
+                else if (nv_arg != nv_arg) {
+                    sv_setpvf(keysv, "%" NVgf, nv_arg);
```

Comment:
I don't quite like this as is, because it remains possible (however unlikely) to construct a floating-point number whose bytes happen to be all ASCII digits:

```
eval: pack "F", 6.82132005170133e-38
'62345678'
```

and thus they'd compare equal using this algorithm. Can you prepend or suffix a non-digit, perhaps "r" for "raw", in order to avoid this?